main.go

```diff
@@ -155,6 +158,14 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "GCPCluster")
 		os.Exit(1)
 	}
+	if err = (&expcontrollers.GCPManagedControlPlaneReconciler{
```

Comment:
Could we remove this? And then we can re-add the controller creation to main when we add the feature flag in a later PR. The reason being we don't want any of the functionality to be enabled unless there is a feature flag and we should do that after we have merged the controller implementations.

Reply:
makes sense! removed